pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -72,24 +77,39 @@ func (t *Terminator) Taint(ctx context.Context, node *corev1.Node) error {
 		if err := t.kubeClient.Patch(ctx, node, client.StrategicMergeFrom(stored)); err != nil {
 			return err
 		}
-		log.FromContext(ctx).Info("tainted node")
+		taintValues := []any{
+			"taint.Key", taint.Key,
+			"taint.Value", taint.Value,
+		}
+		if len(string(taint.Effect)) > 0 {
+			taintValues = append(taintValues, "taint.Effect", taint.Effect)
+		}
+		log.FromContext(ctx).WithValues(taintValues...).Info("tainted node")
 	}
 	return nil
 }
 
 // Drain evicts pods from the node and returns true when all pods are evicted
 // https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
-func (t *Terminator) Drain(ctx context.Context, node *corev1.Node) error {
+func (t *Terminator) Drain(ctx context.Context, node *corev1.Node, nodeGracePeriodExpirationTime *time.Time) error {
 	pods, err := nodeutil.GetPods(ctx, t.kubeClient, node)
 	if err != nil {
 		return fmt.Errorf("listing pods on node, %w", err)
 	}
+
+	podsToDelete := lo.Filter(pods, func(p *corev1.Pod, _ int) bool {
+		return podutil.IsWaitingEviction(p, t.clock) && !podutil.IsTerminating(p)
+	})
+	if err := t.DeleteExpiringPods(ctx, podsToDelete, nodeGracePeriodExpirationTime); err != nil {
+		return err
```

Comment:
Can you wrap this error? 
```suggestion
		return fmt.Errorf("deleting expiring pods, %w", err)
```